src/extension/background-script/actions/ln/request.ts

```diff
@@ -0,0 +1,82 @@
+import utils from "~/common/lib/utils";
+import { MessageGenericRequest } from "~/types";
+
+import db from "../../db";
+import state from "../../state";
+
+const request = async (
+  message: MessageGenericRequest
+): Promise<{ data: unknown } | { error: string }> => {
+  const connector = await state.getState().getConnector();
+
+  const { origin, args } = message;
+
+  try {
+    if (!connector.requestMethod) {
+      throw new Error(
+        `${message.args.method} is not supported by your account`
+      );
+    }
+
+    const allowance = await db.allowances
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .first();
+
+    if (!allowance?.id) {
+      return { error: "Could not find an allowance for this host" };
+    }
+
+    const permission = await db.permissions
+      .where("host")
+      .equalsIgnoreCase(origin.host)
+      .and((p) => p.method === args.method)
+      .first();
+
+    if (permission && permission.enabled) {
+      const response = await connector.requestMethod(args.method, args.params);
```

Comment:
Not sure how this will improve the readability?

Right now, there is an early exit if the request method can be called immediately.
If not, do this and that (long code block).